### PR TITLE
Fix CI: ignore C901 ("some_func" is to complex) in `ruff`

### PR DESCRIPTION
**Pull request by polinaeterna**
idk if I should have added this ignore to `ruff` too, but I added :)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 [tool.black]
 line-length = 119
 target_version = ['py37']
+ignore = ["C901"]
 
 [tool.ruff]
 # Ignored rules:
 #   "E501" -> line length violation
 #   "F821" -> undefined named in type annotation (e.g. Literal["something"])
```
Comment (lhoestq) [inline pyproject.toml:9]: maybe add why we ignore it ?

**Review by lhoestq — APPROVED** [review]
thanks for the fix !

File: pyproject.toml [hunk 2]
```diff
@@ -1,12 +1,13 @@
 [tool.black]
 line-length = 119
 target_version = ['py37']
+ignore = ["C901"]
```
Comment (albertvillanova) [inline pyproject.toml:4]: Are you sure we need it here as well?

**Review by albertvillanova — APPROVED** [review]
Thanks for the fix.

File: pyproject.toml (hunk 2 above)
Comment (polinaeterna) [reply pyproject.toml]: ah true, thanks! I first thought it was `black` that triggers this error.

**Pull request merged**